keras_cv/metrics/coco/recall.py

```diff
@@ -12,11 +12,185 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 import tensorflow as tf
+import tensorflow.keras as keras
+import tensorflow.keras.initializers as initializers
 
-from keras_cv.metrics.coco.base import COCOBase
+from keras_cv.metrics.coco import utils
+from keras_cv.utils import bbox
+from keras_cv.utils import iou as iou_lib
 
 
-class COCORecall(COCOBase):
+class COCORecall(keras.metrics.Metric):
+    """COCORecall computes the COCO recall metric.
+
+    Args:
+        class_ids: The class IDs to evaluate the metric for.  To evaluate for
+            all classes in over a set of sequentially labelled classes, pass
+            `range(num_classes)`.
+        iou_thresholds: IoU thresholds over which to evaluate the recall.  Must
+            be a tuple of floats, defaults to [0.5:0.05:0.95].
+        area_range: area range to constrict the considered bounding boxes in
+            metric computation. Defaults to `None`, which makes the metric
+            count all bounding boxes.  Must be a tuple of floats.  The first
+            number in the tuple represents a lower bound for areas, while the
+            second value represents an upper bound.  For example, when
+            `(0, 32**2)` is passed to the metric, recall is only evaluated for
+            objects with areas less than `32*32`.  If `(32**2, 1000000**2)` is
+            passed the metric will only be evaluated for boxes with areas larger
+            than `32**2`, and smaller than `1000000**2`.
+        max_detections: number of maximum detections a model is allowed to make.
+            Must be an integer, defaults to `100`.
+    Usage:
+
+    COCORecall accepts two Tensors as input to it's `update_state` method.
+    These Tensors represent bounding boxes in `corners` format.  Utilities
+    to convert Tensors from `xywh` to `corners` format can be found in
+    `keras_cv.utils.bbox`.
+
+    Each image in a dataset may have a different number of bounding boxes,
+    both in the ground truth dataset and the prediction set.  In order to
+    account for this, you may either pass a `tf.RaggedTensor`, or pad Tensors
+    with `-1`s to indicate unused boxes.  A utility function to perform this
+    padding is available at `keras_cv_.utils.bbox.pad_bbox_batch_to_shape`.
+
+    ```python
+    coco_recall = keras_cv.metrics.coco.COCORecall(
```

Comment:
From the __init__, I guess the input API is keras_cv.metrics.COCORecall.

Reply:
Thanks - good catch.